README.md

```diff
@@ -140,9 +144,121 @@ In another terminal run the e2e tests
+
+Files for the local database e2e-tests are located in the `./end-to-end-test/local` directory of cbioportal-frontend. The directory structure of `./end-to-end-test/local` is comparable to that of the `./end-to-end-test/remote` directory used for e2e-tests against remote public cBioPortal instances.
+
+### Contexts for e2e-localdb tests
+E2e-tests can run in _local context_ (workstation for software development) or _CircleCI context_ (for continuous integration testing). On CircleCI, e2e-tests can be conducted on commits that are inside or outside the context of a pull request. These different contexts for _e2e-localdb_ tests are refered to below as _Local_, _CircleCi_ and _CircleCI+PR_ contexts, respectively.
+
+### Running e2e-localdb tests in _Local_ context for development
+Running of e2e-localdb tests in _Local_ context in essence follows the procedure as defined in `.circleci/config.yml`. In _Local_ context the e2e-localdb tests can be run in a dockerized environment (usefull for generating screenshots) or directly on the host sytem. The `end-to-end-test/local/runtime-config/run_container_screenshot_test.sh` script is included to run e2e-local tests in a dockerized environment in _Local_ context. The `end-to-end-test/local/runtime-config/run_local_screenshot_test.sh` is included to run e2e-local tests on the host system directly in _Local_ context. Prior to running the script several environmental variables should be set. For local development follow these steps:
+
+1. Add `export BACKEND=cbioportal:rc` (or other backend branch) to `/env/custom.sh`.
+2. Make sure the last commit is pushed to _origin_ repo (needed by jitpack).
+3. Setup a local docker container environment: 
+
+```bash
+export PORTAL_SOURCE_DIR=~/git/cbioportal-frontend # change path if needed
+export TEST_HOME="$PORTAL_SOURCE_DIR/end-to-end-test"
+source $PORTAL_SOURCE_DIR/env/custom.sh
+cd $TEST_HOME/local/runtime-config
+eval "$(./setup_environment.sh)"
+cd $PORTAL_SOURCE_DIR
+```
+
+4. Running tests can be executed (a) on the local system or (b) in the docker environment;
+
+(a) local system (best for test development)
+```bash
+export CBIOPORTAL_URL='http://localhost:8081'
+$TEST_HOME/local/runtime-config/setup_local_context.sh -j -d -p -e # remove flags to exclude specific stages if desired (see below)
+cd $PORTAL_SOURCE_DIR
+./end-to-end-test/local/runtime-config/run_local_screenshot_test.sh # (repeat this step while developing)
+```
+
+(b) docker environment (best for making reference screenshots)
+```bash
+export CBIOPORTAL_URL='http://cbioportal:8080'
+yarn
+yarn build
+cd $TEST_HOME/local/docker
+./setup_docker_containers.sh -p -d -e # remove flags to exclude specific stages if desired (see below)
+cd $PORTAL_SOURCE_DIR
+$TEST_HOME/local/runtime-config/run_container_screenshot_test.sh
+```
+
+5. When finished reclaim ports by killing selenium and node servers:
+```bash
+killall java -9 # reclaims port 4444
+killall node -9 # reclaims port 3000
```

Comment:
@pvannierop potentially dangerous if running other java or node apps on the machine, but not that big of a deal

Reply:
Ok, I changed the instructions to specifically killing the 3000 and 4444 ports:
```
lsof -n -i4TCP:4444 && lsof -n -i4TCP:4444 | grep LISTEN | awk '{ print $2 }' | xargs kill # reclaims port 4444
lsof -n -i4TCP:3000 && lsof -n -i4TCP:3000 | grep LISTEN | awk '{ print $2 }' | xargs kill # reclaims port 3000
```